spring-xd-module/src/main/java/org/springframework/xd/module/spark/SparkStreamingDriverModule.java

```diff
@@ -0,0 +1,329 @@
+				}
+			}
+		});
+		try {
+			boolean started = receiverStartLatch.await(30, TimeUnit.SECONDS);
+			if (!started) {
+				logger.warn("Deployment timed out when deploying Spark Streaming module " + processor);
+			}
+			if (!receiverStartSuccess.get()) {
+				throw new IllegalStateException("Failed to start Spark Streaming Receiver");
+			}
+		}
+		catch (InterruptedException ie) {
+			throw new RuntimeException(ie);
+		}
+	}
+
+	/**
+	 * Retrieve spark configuration properties from the {@link org.springframework.xd.spark.streaming.Processor} implementation.
+	 * This method uses {@link SparkConfig} annotation to derive the {@link Properties} returned
+	 * from the annotated methods.
+	 *
+	 * @param processor the spark streaming processor
+	 * @return the spark configuration properties (if defined) or empty properties
+	 */
+	public static Properties getSparkModuleProperties(Processor processor) {
+		Properties sparkConfigs = new Properties();
+		Method[] methods = processor.getClass().getDeclaredMethods();
+		for (Method method : methods) {
+			SparkConfig sparkConfig = method.getAnnotation(SparkConfig.class);
+			if (sparkConfig != null) {
+				try {
+					if (method.getReturnType().equals(Properties.class)) {
+						sparkConfigs.putAll((Properties) method.invoke(processor));
+					}
+					else {
+						logger.warn("@SparkConfig annotated method should return java.util.Properties type. " +
+								"Ignoring the method " + method.getName());
+					}
+				}
+				catch (InvocationTargetException ise) {
+					// ignore.
+				}
+				catch (IllegalAccessException ise) {
+					// ignore.
+				}
+			}
+		}
+		return sparkConfigs;
+	}
+
+	/**
+	 * Get the list of jars that this spark module requires.
+	 *
+	 * @return the list of spark application jars
+	 */
+	private List<String> getApplicationJars() {
+		// Get jars from module classpath
+		URLClassLoader classLoader = (URLClassLoader) this.getClassLoader();
+		List<String> jars = new ArrayList<String>();
+		for (URL url : classLoader.getURLs()) {
+			String file = url.getFile().split("\\!", 2)[0];
+			if (file.endsWith(".jar")) {
+				jars.add(file);
+			}
+		}
+		// Get message bus libraries
+		Environment env = this.getApplicationContext().getEnvironment();
+		String jarsLocation = env.resolvePlaceholders(MESSAGE_BUS_JARS_LOCATION);
+		try {
+			Resource[] resources = resolver.getResources(jarsLocation);
+			for (Resource resource : resources) {
+				URL url = resource.getURL();
+				jars.add(url.getFile());
+			}
+		}
+		catch (IOException ioe) {
+			throw new RuntimeException(ioe);
+		}
+		// Get necessary dependencies from XD DIRT.
+		URLClassLoader parentClassLoader = (URLClassLoader) this.getClassLoader().getParent();
+		URL[] urls = parentClassLoader.getURLs();
+		for (URL url : urls) {
+			String file = url.getFile().split("\\!", 2)[0];
+			//if (file.endsWith(".jar")) {
+			if (file.endsWith(".jar") && (// Add spark jars
+					file.contains("spark") ||
```

Comment:
yeah, agreed ;) but the fact that this significantly reduced the number of jars sent across to spark executor running on the outside cluster. This code still can be optimized once we have a simple way to setup message bus configuration (for MessageBusReceiver/Sender) on the spark executor side.